src/main/java/com/google/step/servlets/JsonFormatter.java

```diff
@@ -214,4 +214,58 @@ public static String getUserJson(
   private static String getImageUrl(String blobKey) {
     return "/api/images/" + blobKey;
   }
+
+  /**
+   * Returns a json of home page data.
+   *
+   * @param homePageDealsMaps a list of list of maps for home page
+   * @return a json of home page data
+   */
+  public static String getHomePageJson(List<List<Map<String, Object>>> homePageDealsMaps) {
+    Gson gson = new Gson();
+    String json = gson.toJson(getHomePageMap(homePageDealsMaps));
+    return json;
+  }
+
+  /**
+   * Returns a map of sections for home page
+   *
+   * @param homePageDealsMaps a list of list of maps for home page
+   * @return a map of sections for home page
+   */
+  private static Map<String, Object> getHomePageMap(
+      List<List<Map<String, Object>>> homePageDealsMaps) {
+    Map<String, Object> homePageMap = new HashMap<>();
+    homePageMap.put("popularDeals", homePageDealsMaps.get(0));
+    homePageMap.put("usersIFollow", homePageDealsMaps.get(1));
+    homePageMap.put("restaurantsIFollow", homePageDealsMaps.get(2));
+    homePageMap.put("tagsIFollow", homePageDealsMaps.get(3));
+    return homePageMap;
+  }
+
+  /**
+   * Returns a map of brief deal info for home page
+   *
+   * @param deal the deal to create a map for
+   * @param poster the user object of the poster
+   * @param restaurant restaurant the deal is at
+   * @param tags tags associated with the deal
+   * @param votes number of votes the deal received
+   * @return a map of brief deal info for home page
+   */
+  public static Map<String, Object> getBriefHomePageDealMap(
```

Comment:
Use `getBriefDealMap` instead. Update the `getBriefDealMap` arguments to use whatever you need, and update the places that call `getBriefDealMap` as well. Brief Deal Map don't need tags I think.

Reply:
is this still relevant since we are removing the restaurant name/poster name and votes attribute for brief deal map

Reply:
ok you can keep this method, but add a wrapper that transforms this into a Json String

Reply:
actually i think you can leave it as a method that returns a public map. Then you don't have to manually do string concatenation `"[" + json + "," + ... + "]"`